sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -949,14 +949,19 @@ class Analyzer(
             if oldVersion.outputSet.intersect(conflictingAttributes).nonEmpty =>
           (oldVersion, oldVersion.copy(serializer = oldVersion.serializer.map(_.newInstance())))
 
-        // Handle projects that create conflicting aliases.
         case oldVersion @ Project(projectList, _)
-            if findAliases(projectList).intersect(conflictingAttributes).nonEmpty =>
-          (oldVersion, oldVersion.copy(projectList = newAliases(projectList)))
+            if hasConflict(projectList, conflictingAttributes) =>
+          (oldVersion,
+            oldVersion.copy(
+              projectList =
+                newNamedExpression(projectList, conflictingAttributes)))
 
         case oldVersion @ Aggregate(_, aggregateExpressions, _)
-            if findAliases(aggregateExpressions).intersect(conflictingAttributes).nonEmpty =>
-          (oldVersion, oldVersion.copy(aggregateExpressions = newAliases(aggregateExpressions)))
+            if hasConflict(aggregateExpressions, conflictingAttributes) =>
+          (oldVersion,
+            oldVersion.copy(
+              aggregateExpressions =
+                newNamedExpression(aggregateExpressions, conflictingAttributes)))
```

Comment:
Could you make the above 4 lines simpler like the following?
```scala
-          (oldVersion,
-            oldVersion.copy(
-              aggregateExpressions =
-                newNamedExpression(aggregateExpressions, conflictingAttributes)))
+          val newExprs = newNamedExpression(aggregateExpressions, conflictingAttributes)
+          (oldVersion, oldVersion.copy(aggregateExpressions = newExprs))
```

Reply:
updated as suggested. thanks!